### PR TITLE
Phone Home: Fix found issues on real deployment

### PR DESCRIPTION
**Pull request by sgnn7**
https://phabricator.endlessm.com/T12342

### CODE REVIEW

File: activate [hunk 1]
```diff
@@ -27,7 +27,14 @@
 DIR="/var/lib/eos-phone-home"
 if [ ! -d "${DIR}" ] || [ ! -w "${DIR}" ]; then
     echo "${DIR} missing or not writable"
-    exit 1
+
+    if [ "$EUID" == "0" ]; then
+      echo "Making ${dir}"
+      mkdir -p /var/lib/eos-phone-home
```
Comment (ramcq) [inline activate]: Hmm... creating this is supposed to be covered by the systemd tmpfiles entry - is that not working in some situation?

Comment (sgnn7) [inline activate]: Makes it work locally with `sudo` on a dev machine and won't hurt real deployments

File: activate [hunk 2]
```diff
@@ -71,25 +82,45 @@ serial="${serial%"${serial##*[![:space:]]}"}"
 # detect live system
 live="false"
 if grep -qw "endless.live" /proc/cmdline; then
-    image="true"
+    live="true"
+fi
+
+
+# We don't want dummy serials sent
+if [ "${serial}" == "To be filled by O.E.M." ]; then
```
Comment (ramcq) [inline activate]: We already have these kind of bogus entries in the metrics system from OEMs who have missed completing things - in all three fields from DMI we can get "To Be Filled By O.E.M." (actually more popular) or "To be filled by O.E.M.", and one of our 6th most popular make/models is the award-winning "System Product Name" from "System manufacturer". What I'm saying is bogus information here is a fact of life and I think trying to sanitise it is a losing battle. I don't think we want to get into trying to filtering on the client - we can just deal with this on the server by permitting these to be non-unique and assuming every activation is a potentially relevant event and storing it anyway - then if we do need to analyse them and discard bogus stuff or de-duplicate we can do it on the aggregate data-set.

File: activate [hunk 3]
```diff
@@ -71,25 +82,45 @@ serial="${serial%"${serial##*[![:space:]]}"}"
...
+
+print json.dumps(activation)
+END
+)
+
+echo "Sending: ${json}"
 
 # report in
 if ! out=$(wget -O - -q --method=PUT --body-data="${json}" "${SCRIPT}") ; then
     echo "failed PUT to $SCRIPT with data: ${json}"
```
Comment (ramcq) [inline activate]: let's not echo the JSON blob twice if you're going to log it unconditionally above

File: activate [hunk 4]
```diff
@@ -71,25 +82,45 @@ serial="${serial%"${serial##*[![:space:]]}"}"
...
     exit 1
 fi
-if ! ( echo "${out}" | python -c "import json, sys; sys.exit(0 if json.load(sys.stdin)['success'] == 'true' else 1)" 2>/dev/null ); then
+if ! ( echo "${out}" | python -c "import json, sys; sys.exit(0 if json.load(sys.stdin)['success'] == True else 1)" 2>/dev/null ); then
     echo "didn't get success from PUT to ${SCRIPT} with data: ${json}"
```
Comment (ramcq) [inline activate]: similarly

File: activate (hunk 2 above)
Comment (sgnn7) [inline activate]: Serial is the important one so we don't reprocess the same data over and over again including sending to asus but if that's not an issue, I'll take that out.

File: activate (hunk 3 above)
Comment (sgnn7) [inline activate]: will fix

File: activate (hunk 4 above)
Comment (sgnn7) [inline activate]: ditto

File: activate (hunk 2 above)
Comment (ramcq) [inline activate]: Meh? I doubt they can complain at us very convincingly if they didn't fill their BIOS in correctly... :) We're not going to call them back unless it's an eosoem image, so it should only be the ASUS systems they put EOS onto that do this.

File: activate [hunk 5]
```diff
@@ -71,25 +82,45 @@ serial="${serial%"${serial##*[![:space:]]}"}"
...
+activation["live"] = os.environ["live"] == "true"
+
+print json.dumps(activation)
+END
+)
+
+echo "Sending: ${json}"
 
 # report in
 if ! out=$(wget -O - -q --method=PUT --body-data="${json}" "${SCRIPT}") ; then
```
Comment (sgnn7) [inline activate]: @ramcq This isn't going to work - wget keeps sending `<json_data>=` to the server so it fails json validation and curl isn't installed by default. I think I'll try rewriting this thing in Python from scratch.

**Comment by sgnn7** [comment]
Closing for rewrite in python

**Pull request closed without merging**